src/transformers/models/bart/configuration_bart.py

```diff
@@ -180,30 +182,171 @@ def __init__(
+                )
+
+                common_inputs["past_key_values"] = []
+                # If the number of encoder and decoder layers are present in the model configuration, both are considered
+                num_encoder_layers, num_decoder_layers = self.num_layers
+                min_num_layers = min(num_encoder_layers, num_decoder_layers)
+                max_num_layers = max(num_encoder_layers, num_decoder_layers) - min_num_layers
+                remaining_side_name = "encoder" if num_encoder_layers > num_decoder_layers else "decoder"
+
+                for _ in range(min_num_layers):
+                    common_inputs["past_key_values"].append(
+                        (
+                            torch.zeros(decoder_shape),
+                            torch.zeros(decoder_shape),
+                            torch.zeros(encoder_shape),
+                            torch.zeros(encoder_shape),
+                        )
+                    )
+
+                # TODO: test this.
+                shape = encoder_shape if remaining_side_name == "encoder" else decoder_shape
+                for _ in range(min_num_layers, max_num_layers):
+                    common_inputs["past_key_values"].append((torch.zeros(shape), torch.zeros(shape)))
+
+        elif self.task in ["causal-lm", "causal-lm-with-past"]:
+            common_inputs = super(OnnxConfigWithPast, self).generate_dummy_inputs(
+                tokenizer, batch_size, seq_length, is_pair, framework
+            )
+
+            if self.use_past:
+                if not is_torch_available():
+                    raise ValueError("Cannot generate dummy past_keys inputs without PyTorch installed.")
+                else:
+                    import torch
+
+                    batch = common_inputs["input_ids"].shape[0]
+                    num_encoder_layers, _ = self.num_layers
+                    num_encoder_attention_heads, _ = self.num_attention_heads
+                    past_shape = (
+                        batch,
+                        num_encoder_attention_heads,
+                        1,
+                        self._config.hidden_size // num_encoder_attention_heads,
+                    )
+
+                    common_inputs["attention_mask"] = torch.cat(
+                        [common_inputs["attention_mask"], torch.ones(batch, 1)], dim=1
+                    )
+                    common_inputs["past_key_values"] = [
+                        (torch.zeros(past_shape), torch.zeros(past_shape)) for _ in range(num_encoder_layers)
+                    ]
+        else:
+            common_inputs = super(OnnxConfigWithPast, self).generate_dummy_inputs(
+                tokenizer, batch_size, seq_length, is_pair, framework
+            )
+
+        return common_inputs
+
+    def _flatten_past_key_values_(self, flattened_output, name, idx, t):
+        if self.task in ["default", "default-with-past", "seq2seq-lm", "seq2seq-lm-with-past"]:
```

Comment:
Yes, because you can have `default` and `self.use_past = True` (when you are doing `default-with-past`).
As mentioned above, "default-with-past" will be parsed to something like `OnnxConfig(task=feature.replace("-with-past", ""), use_past="with-past" in feature)`.

Reply:
Perfect, thank you for the clarification!